src/java/htsjdk/samtools/util/SamLocusIterator.java

```diff
@@ -142,6 +142,14 @@ public void add(final SAMRecord read, final int position) {
      */
     private boolean emitUncoveredLoci = true;
 
+    /**
+     * If set, this will cap the number of reads we accumulate for any given position.
+     * Note that if we hit the maximum threshold at the first position in the accumulation queue,
+     * then we throw further reads overlapping that position completely away (including for subsequent positions).
+     * This is a useful feature if one wants to minimize the memory footprint in files with a few massively large pileups.
+     */
+    private long maxReadsToAccumulatePerLocus = -1L;
```

Comment:
Two quick things:
1. Since this is compared to `list.size()` which is int, why not just make it an `int`?
2. Defaulting to `Integer.MAX_VALUE` would give the same default behavior and simplify the check below